tests/system/python/e2e/test_e2e_modbus_c_pi.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test system/python/e2e/test_e2e_modbus_c_pi.py
+
+"""
+
+import http.client
+import json
+import time
+import pytest
+import socket
+import utils
+
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+SOUTH_PLUGIN = "modbus-c"
+PLUGIN_NAME = "ModbusC"
+SVC_NAME = "modbus-c"
+SOUTH_PLUGIN_LANGUAGE = "cpp"
+ASSET_NAME = "A15"
+
+
+class TestE2EModbusCPI:
+    def check_connect(self, modbus_host, modbus_port):
+        s = socket.socket()
+        print("Host is {}, Port is {}".format(modbus_host, modbus_port))
+        result = s.connect_ex((modbus_host, modbus_port))
+        if result != 0:
+            print("Socket connection fail!!")
+            pytest.skip("Test requires a running simulator, please run before starting the test!!")
+
+    def get_ping_status(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/ping')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        return jdoc
+
+    def get_statistics_map(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/statistics')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        return utils.serialize_stats_map(jdoc)
+
+    @pytest.fixture
+    def start_south_north(self, reset_and_start_foglamp, add_south, remove_directories, south_branch, foglamp_url,
+                          start_north_pi_server_c, pi_host, pi_port, pi_token, modbus_host, modbus_port):
+        """ This fixture clone a south and north repo and starts both south and north instance
+
+            reset_and_start_foglamp: Fixture that resets and starts foglamp, no explicit invocation, called at start
+            add_south: Fixture that adds a south service with given configuration with enabled or disabled mode
+            remove_directories: Fixture that remove directories created during the tests
+        """
+        self.check_connect(modbus_host, modbus_port)
+        cfg = {"protocol": {"value": "TCP"}, "asset": {"value": ""}, "port": {"value": "502"},
```

Comment:
is this `"asset": {"value": ""}` required?

Reply:
Yes, else takes the default value "modbus". We want to avoid giving the default asset name value as these will be provided by the asset map.